sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java

```diff
@@ -650,10 +650,39 @@ private Tags createRequestTags(
                     serviceEndpoint != null ? escape(serviceEndpoint) : "NONE"));
             }
 
+            String effectiveServiceAddress = serviceAddress != null ? escape(serviceAddress) : "NONE";
             if (metricTagNames.contains(TagName.ServiceAddress)) {
                 effectiveTags.add(Tag.of(
                     TagName.ServiceAddress.toString(),
-                    serviceAddress != null ? escape(serviceAddress) : "NONE"));
+                    effectiveServiceAddress));
+            }
+
+            boolean containsPartitionId = metricTagNames.contains(TagName.PartitionId);
+            boolean containsReplicaId = metricTagNames.contains(TagName.ReplicaId);
+            if (containsPartitionId || containsReplicaId) {
+
+                String partitionId = "NONE";
+                String replicaId = "NONE";
+
+                if (serviceAddress != null) {
+                    String[] serviceAddressParts = effectiveServiceAddress.split("/");
+                    if (serviceAddressParts.length == 9) {
```

Comment:
Can we put a sample here - to show how this serviceAddressParts array looks like (after splitting or pre-splitting) - probably easier to understand and debug the code if needed.

Reply:
Done